### DISCUSSION TITLE
FS2020 StreamDeck - switch with no hold function

### DISCUSSION THREAD

**Discussion by rms59** [nguyenquyhy/Flight-Tracker-StreamDeck#330] [Q&A] [answered]
Is there a toggle switch for streamdeck (15)  that is a simple toggle without the "press and hold" function? In FS2020 I find I inadvertantly hold a button too long and the states switch which really messes me up.  I just want to get rid of that function.

**Comment by nguyenquyhy** [comment, marked as answer]
May I check which button are you talking about? By default most buttons doesn't repeat when you hold them (with the exception of Increase/Decrease buttons).

**Reply by rms59** [reply]
Hi there. Thanks for the reply.  The problem is Microsoft Flight Simulator 2020 using "toggle" commands in some of their actions. For example command CTRL-W extends the water rudder and CTRL-W again to retract them.  My problem is that if I map any switch on StreamDeck for CTRL-W to extend the water rudders, I can inadvertently hold the switch too long and cause them to retract again which reverses the switch icon indicators. Now I have to visually look at the rudders to see which state there in and ignore the icons. 

I'm not sure if there's a way but I need a switch that will close for a few milliseconds to issue a command then open even if the switch is still being held. This will get rid of the repeat.

**Reply by nguyenquyhy** [reply]
What you are using is Stream Deck feature to trigger a keybind, which has nothing to do with this plugin, so we can't change that.

You can try to use buttons from this Flight Tracker plugin instead. As I said before most of our buttons don't repeat the key on hold by default.

**Reply by rmroc451** [reply]
To @nguyenquyhy 's point, add a toggle button from this app on the stream deck with this as the toggle: `TOGGLE_WATER_RUDDER`

And as a display value you can use `WATER_RUDDER_HANDLE_POSITION`

Alternatively you could change the button background image based on what the handle position could be.

If I have a few minutes today, I'll try it on the icon a5 and show you here.

**Reply by rms59** [reply]
For some reason my reply didn't get through. I don't understand why but I'm new to Github so I'll reply again.
To answer @nguyenguyhy, I was using the hotkey switch but changed it to the Generic Toggle Button in Flight Tracker and that worked and eliminated the repeat problem but doesn't give me any indication what state the rudders are in. If I try to embed or link an ON/OFF image in the appropriate place they are both ignored. The icon images are all in the same folder as the StreamDeck files and my other buttons like AP HOLD work fine. Hoping you and rmroc451 can help me with this. Much appreciated.

**Reply by nguyenquyhy** [reply]
For water rudder, have you tried to put `WATER RUDDER HANDLE POSITION` into Feedback Value setting of the Generic Toggle Button?
![image](https://user-images.githubusercontent.com/1560418/150704164-872e347c-148e-400f-ae68-f7c75ae3fb77.png)

If you have already done so, please export the profile and attach it here, and please also let us know which aircraft you are testing the button on.

**Reply by rms59** [reply]
Seriously thank you... guys. This old timer learned a lot. Using the Generic Toggle Button and the WATER_RUDDER_HANDLE_POSITION for the feedback worked. I now have a button with a green bar when the rudders are down and no bar when they're up and nothing out of sync anymore.
This made my day. 
Richard.
